moveit_core/collision_detection/include/moveit/collision_detection/collision_common.h

```diff
@@ -95,6 +107,15 @@ struct Contact
 
   /** \brief The type of the second body involved in the contact */
   BodyType body_type_2;
+
+  /** \brief The time until collision */
+  double cc_time;
```

Comment:
Time is an overloaded word, especially in terms of trajectories (is it "geometric" time, where the robot moves at the same speed for the entire traj, or trajectory time, where each waypoint is time parameterized? I actually think this is from the range 0 to 1, correct?).  I think the better term would be `cc_percent_distance`, or `collision_percent_completed` (if 0 to 1) or just `cc_distance` if it's actually a distance.

Reply:
Yes, it is in the range of 0 to 1, but I have a hard time understanding what it actually is from [this](https://github.com/j-petit/moveit/blob/4797bc41755e4143d77cb289c29a87e42a087476/moveit_core/collision_detection_bullet/include/moveit/collision_detection_bullet/tesseract/bullet_utils.h#L494) piece of code. Unfortunately, the tesseract code is lacking documentation [here](https://github.com/ros-industrial-consortium/tesseract/blob/740e6607423e2afbeba04b609e118043c86b7266/tesseract_core/include/tesseract_core/basic_types.h#L208)